eval/topdown.go

```diff
@@ -173,6 +173,51 @@ func dereferenceVar(v opalog.Var, ctx *TopDownContext) (interface{}, error) {
 	return valueToInterface(binding, ctx)
 }
 
+func evalContext(ctx *TopDownContext, iter TopDownIterator) error {
+
+	if ctx.Index >= len(ctx.Query) {
+		return iter(ctx)
+	}
+
+	if ctx.Current().Negated {
+		return evalContextNegated(ctx, iter)
```

Comment:
See comments below about safety.  If we're assuming that all vars need to be ground before computing the negation, we should add an assert here for isground() (maybe eventually conditioning that on a compiler-flag).